app/infrastructure/services.py

```diff
@@ -54,6 +72,16 @@ def find_one(self, dataset_id: str) -> Optional[Dataset]:
         except TypeError:
             return None
 
+    @staticmethod
+    def format_filters(filters):
```

Comment:
I think we're missing the special case of `tag` filter. We can either change `tags` to `tag` in consumer or treat it in `format_filters` here